[16.0][MIG] - product_price_category

sbejaoui opened the pull request.

grindtildeath committed:
[add] Module product price category

jcoux committed:
[FIX] Module product price category

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

[UPD] Update product_price_category.pot

[ADD] icon.png

CeciliaOoops committed:
Translated using Weblate (Italian)

Currently translated at 73.3% (11 of 15 strings)

Translation: sale-workflow-10.0/sale-workflow-10.0-product_price_category
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-10-0/sale-workflow-10-0-product_price_category/it/

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (15 of 15 strings)

Translation: sale-workflow-10.0/sale-workflow-10.0-product_price_category
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-10-0/sale-workflow-10-0-product_price_category/ca/

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (15 of 15 strings)

Translation: sale-workflow-10.0/sale-workflow-10.0-product_price_category
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-10-0/sale-workflow-10-0-product_price_category/es/

[UPD] Switch to Github Actions / Copier update

OCA-git-bot committed:
product_price_category 10.0.1.0.1

sbejaoui committed:
[16.0][MIG] - product_price_category

[FIX] - fix ui

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

flotho commented on product_price_category/views/product_template.xml:
```diff
@@ -0,0 +1,23 @@
+        <field name="inherit_id" ref="product.product_template_form_view" />
+        <field name="arch" type="xml">
+            <field name="list_price" position="after">
+                <field name="price_category_id" />
```
Small comment here,
When there is no price category, the field is not visible :+1: 
![image](https://user-images.githubusercontent.com/1043935/211087901-0c328ac6-5ee1-4957-956f-f089dc8a1981.png)

Any chance to add a label?

flotho reviewed (changes requested):
small changes requested on view, otherwise LGTM

lmignon commented on product_price_category/views/product_template.xml:
```diff
@@ -0,0 +1,23 @@
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_form_view" />
+        <field name="arch" type="xml">
+            <field name="list_price" position="after">
+                <field name="price_category_id" />
+            </field>
```
```suggestion
            <div name="standard_price_uom" position="after">
                <field name="price_category_id" />
            </div>
```

lmignon reviewed (changes requested):
@sbejaoui LGTM, just a little change required to fix the UI (Code review + functional tests)

rousseldenis commented:
/ocabot migration product_price_category

rousseldenis commented:
@sbejaoui Don't forget to do pre-commit/formatting stuffs in another commit in order to have migration related stuffs only in migration commit. Follow https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

(pre-commit stuff commit with --no-verify command)

rousseldenis commented on product_price_category/models/product_price_category.py:
```diff
@@ -0,0 +1,12 @@
+
+class ProductPriceCategory(models.Model):
+    _name = "product.price.category"
+    _description = "ProductPriceCategory"
```
```suggestion
    _description = "Product Price Category"
```

rousseldenis commented on product_price_category/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,39 @@
+        is not price_category
+        """
+        if self.applied_on != "2b_product_price_category":
+            self.price_category_id = False
```
Shouldn't be the 'price_category_id' field converted into computed field readonly False in order to remove onchange?

rousseldenis commented on product_price_category/tests/test_sale.py:
```diff
@@ -0,0 +1,132 @@
+
+
+class TestSale(TransactionCase):
+    def setUp(self):
```
Could you convert this into class method ?

rousseldenis reviewed (changes requested):
Code review

sbejaoui commented:
can you check @rousseldenis

rousseldenis reviewed (approved):
Code review

lmignon reviewed (commented):
LGTM (code + functional)

flotho reviewed (approved):
thanks for the change
LGTM

marielejeune commented on product_price_category/README.rst:
```diff
@@ -0,0 +1,69 @@
+
+In Pricelist Form (Sales -> Configuration -> Pricelist), you can add or modify
+an item ("Manage Pricelist Items" access should be checked in the user settings)
+and select "Price Category" in "Applied on".
```
"Advanced price rules (discounts, formulas)" pricelist option must be activated to enable this feature, maybe it is better to add this info in the README?

rousseldenis replied:
@sbejaoui

marielejeune commented on product_price_category/README.rst:
```diff
@@ -0,0 +1,69 @@
+Usage
+=====
+
+In Pricelist Form (Sales -> Configuration -> Pricelist), you can add or modify
```
Since we need `sale_management` module to configure the items, shouldn't we depend on it in the manifest?

rousseldenis replied:
No, but if we depends on menus defined there

rousseldenis commented:
@sbejaoui

Commit:
[FIX] product_price_category: compute `name` correctly

Prior to this commit `name` was not correctly computed when `applied_on`
value is `2b_product_price_category`.

IT-Ideas commented:
Hello @sbejaoui I just pushed a fix commit as the name was not computed correctly when `applied_on` value is `2b_product_price_category`.

JordiMForgeFlow reviewed (approved):
Code + functional review 👍🏼

rousseldenis commented:
All comments attended 

/ocabot merge nobump
